examples/acoustic/gradient_example.py

```diff
@@ -0,0 +1,176 @@
+import numpy as np
+from examples.seismic import Model
+from devito import Dimension, time, TimeData, DenseData
+from examples.source_type import SourceLike
+from examples.acoustic.fwi_operators import ForwardOperator, GradientOperator
+from numpy import linalg
+
+
+def smooth10(vel, dimensions):
+    out = np.zeros(dimensions)
+    out[:] = vel[:]
+    for a in range(5, dimensions[-1]-6):
+        if len(dimensions) == 2:
+            out[:, a] = np.sum(vel[:, a - 5:a + 5], axis=1) / 10
+        else:
+            out[:, :, a] = np.sum(vel[:, :, a - 5:a + 5], axis=2) / 10
+    return out
+
+
+def source(t, f0):
+    r = (np.pi * f0 * (t - 1./f0))
+    return (1-2.*r**2)*np.exp(-r**2)
+
+
+def run(dimensions=(50, 50, 50), spacing=(15.0, 15.0, 15.0), tn=750.0,
+        time_order=2, space_order=4, nbpml=40, dse='noop', dle='noop',
+        autotuning=False, compiler=None, cache_blocking=None):
+    ndim = len(dimensions)
+    origin = tuple([0.] * ndim)
+    f0 = .010
+
+    t0 = 0.0
+    nsrc = 1
+    nrec = 101
+    # True velocity
+    true_vp = np.ones(dimensions) + .5
+    true_vp[:, :, int(dimensions[2] / 2):] = 2.
+
+    # Smooth velocity - we use this as our initial m
+    initial_vp = smooth10(true_vp, dimensions)
+    m0 = initial_vp**-2
+    # Model perturbation
+    dm = true_vp**-2 - initial_vp**-2
+    model = Model(origin, spacing, true_vp.shape, true_vp, nbpml=nbpml)
+    dt = model.critical_dt
+    if time_order == 4:
+        dt *= 1.73
+    dtype = model.dtype
+    h = model.get_spacing()
+    nt = int(1+(tn-t0)/dt)
+
+    # Source geometry
+    time_series = np.zeros((nt, nsrc))
+    time_series[:, 0] = source(np.linspace(t0, tn, nt), f0)
+
+    location = np.zeros((nsrc, 3))
+    location[0, 0] = origin[0] + dimensions[0] * spacing[0] * 0.5
+    location[0, 1] = origin[1] + dimensions[1] * spacing[1] * 0.5
+    location[0, 2] = origin[1] + 2 * spacing[2]
+
+    # Receiver geometry
+    receiver_coords = np.zeros((nrec, 3))
+    receiver_coords[:, 0] = np.linspace(0, origin[0] + dimensions[0] * spacing[0],
+                                        num=101)
+    receiver_coords[:, 1] = origin[1] + dimensions[1] * spacing[1] * 0.5
+    receiver_coords[:, 2] = location[0, 2]
+
+    # Create source symbol
+    p_src = Dimension('p_src', size=nsrc)
+    src = SourceLike(name="src", dimensions=[time, p_src], npoint=nsrc, nt=nt,
+                     dt=dt, h=h, ndim=ndim, nbpml=nbpml, dtype=dtype,
+                     coordinates=location)
+    src.data[:] = time_series
+
+    # Create receiver symbol
+    p_rec = Dimension('p_rec', size=nrec)
+
+    # Receiver for true model
+    rec_t = SourceLike(name="rec_t", dimensions=[time, p_rec], npoint=nrec, nt=nt, dt=dt,
```

Comment:
If the objective is to not use seismic-specific utilities, should we not rebase this on #239 and use internal `PointData` instead?